### PR TITLE
Adds timeout and docstrings to identity SDK

### LINKED ISSUE #10083: Update identity SDK to abort hanging Promises
accept optional `AbortController` 

refer to https://celo-org.slack.com/archives/C02TX7BGJHG/p1671474844544509

### LINKED ISSUE #10090: Add docstrings to identity SDK

### PR DESCRIPTION
**Pull request by alecps**
### Description

- allows an AbortController to be passed into ODIS PNP requests to enable clients to timeout cleanly
- Adds detailed docstrings to the identity SDK where they were missing

### Other changes

- Adds more identifier prefixes per the request of a partner wallet 
- Makes small updates to the ODIS load test code that were useful for testing 

### Tested

Load tests run against alfajores showed that timeouts (even in large volumes) do not introduce problems in ODIS server-side

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/10083
- Fixes https://github.com/celo-org/celo-monorepo/issues/10090

### Backwards compatibility

Yes

### Documentation

ASv2 docs will be updated to reflect the AbortController parameter

### CODE REVIEW

File: packages/phone-number-privacy/monitor/src/test.ts [hunk 1]
```diff
@@ -35,7 +38,27 @@ export async function testPNPQuery(
...
+    const odisResponse: PnpClientQuotaStatus = await queryOdisForQuota(
+      blockchainProvider,
+      contextName,
+      timeoutMs
+    )
+    logger.info({ odisResponse }, 'ODIS quota request successful. System is healthy.')
+  } catch (err) {
+    logger.error('ODIS quota request failed.')
+    logger.error({ err })
+    throw err
```
Comment (aaronmgdr) [inline packages/phone-number-privacy/monitor/src/test.ts:57]: why rethrow?

File: packages/sdk/identity/src/odis/identifier.ts [hunk 2]
```diff
@@ -24,13 +24,33 @@ const sha3 = (v: string) => soliditySha3({ type: 'string', value: v })
...
+ * i.e. if a user's instagram and twitter handles are the same,
+ * these prefixes prevent the ODIS identifers from being the same.
+ *
+ * If you would like to use a prefix that isn't included, please put up a PR adding it
+ * to ensure interoperability with other projects. When adding new prefixes,
+ * please use either the full platform name in all lowercase (e.g. 'facebook')
+ * or DID methods https://w3c.github.io/did-spec-registries/#did-methods.
+ *
+ * The NULL prefix is included to allow projects to use the sdk without selecting
+ * a predefined prefix or adding there own. Production use of the NULL prefix is
```
Comment (isabellewei) [inline packages/sdk/identity/src/odis/identifier.ts]:
```suggestion
 * a predefined prefix or adding their own. Production use of the NULL prefix is
```

File: packages/sdk/identity/src/odis/identifier.ts [hunk 3]
```diff
@@ -205,11 +265,28 @@ export async function getObfuscatedIdentifierFromSignature(
   }
 }
 
+/**
+ * Concatenates the identifierPrefix and plaintextIdentifier with the seperator '://'
```
Comment (isabellewei) [inline packages/sdk/identity/src/odis/identifier.ts]:
```suggestion
 * Concatenates the identifierPrefix and plaintextIdentifier with the separator '://'
```

**Review by isabellewei — APPROVED** [review]
woohooo thanks for adding the tests and docstrings

File: packages/phone-number-privacy/monitor/src/test.ts (hunk 1 above)
Comment (alecps) [reply packages/phone-number-privacy/monitor/src/test.ts:57]: just so that the monitor cloud function fails

**Pull request merged**